File: 20200215Python-China/0066. Plus One.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        flag=1
        for k in range(len(digits)-1,-1,-1):
            if flag==0:
                break
            if digits[k]==9:
                digits[k]=0
            else:
                digits[k]+=1
                flag=0
        if flag==1:
            return [1]+digits
        return digits

File: archives/C#/0066. Plus One.cs
public class Solution {
    public int[] PlusOne(int[] digits) {
        int flag=1;
        for(int i=digits.Length-1;i>=0;i--){
            int tmp=digits[i]+flag;
            if (tmp>=10){
                flag=1;
                digits[i]=tmp-10;
            }
            else{
                digits[i]=tmp;
                flag=0;
                break;
            }
        }
        
        if (flag==0){
            return digits;
        }
        int[] result=new int[digits.Length+1];
        result[0]=1;
        for (int i=0;i<digits.Length;i++){
            result[i+1]=digits[i];
        }
        return result;
    }
}
